pkg/engine/template_generator.go

```diff
@@ -258,8 +258,8 @@ func (t *TemplateGenerator) getTemplateFuncMap(cs *api.ContainerService) templat
 		"GetCustomCloudIdentitySystem": func() string {
 			return cs.Properties.GetCustomCloudIdentitySystem()
 		},
-		"IsMultipleMasters": func() bool {
-			return cs.Properties.MasterProfile != nil && cs.Properties.MasterProfile.Count > 1
+		"IsMultiMasterCluster": func() bool {
+			return cs.Properties.MasterProfile.HasMultipleNodes()
```

Comment:
This is an example of the business logic being moved out of the funcmap itself and into the unit testable `MasterProfile.HasMultipleNodes()` method.

Reply:
this is called in nodecustomdata so we still need to check for nil Master profile

Reply:
(and in general I think it's always better to nil check MasterProfile everywhere we dereference it since new uses could be added later in code paths that are used by hosted masters)

Reply:
I almost think for defensiveness we should move all `MasterProfile` methods into `Properties` so we can do that nil check in once place.

Reply:
yes also arguably nil master profile (which returns false in this case) is part of the logic and is one of the cases that should be tested for in the unit test and moving the nil check to the function itself would allow us to do that.

Reply:
Yeah, will do that as a discrete PR (apart from these unit test PRs)